test/functional/plugin/tutor_spec.lua

```diff
@@ -0,0 +1,64 @@
+    clear({ args = { '-u', 'NORC' } })
+    command('set cmdheight=0')
+    command('Tutor')
+  end)
+
+  it('should apply transformation', function()
+    local test_table = is_os('win') and {
+      ['!dir'] = {
+        '{2:  } 1. Type :{1:^!}{3:dir} to get a listing of your directory.                            |',
+        '{2:  }    To verify this, type :{1:^!}{3:dir} again to see your directory.                   |',
+        '{2:  } 5. Neovim will write the selected lines to the file TEST. Use :{1:^!}{3:dir} to see it|',
+        '{2:  }        :r {1:^!}{3:dir}                                                               |',
+        '{2:  }     :{1:^!}{3:dir}                   - shows a directory listing                      |',
+        '{2:  } 5. {4::r }{5:^!dir}                  reads the output of the dir command and          |',
+        '{2:  } 1. List the contents of the current directory: :{1:^!}{3:dir}                         |'
+      },
+      ['!del'] = {
+        '{2:  }        :{1:^!}{3:del} TEST                                                            |',
+        '{2:  }     :{1:^!}{3:del} FILENAME          - removes file FILENAME                          |',
+      }
+    } or {
+      ['!ls'] = {
+        '{2:  } 1. Type :{1:^!}{3:ls} to get a listing of your directory.                             |',
+        '{2:  }    To verify this, type :{1:^!}{3:ls} again to see your directory.                    |',
+        '{2:  } 5. Neovim will write the selected lines to the file TEST. Use :{1:^!}{3:ls} to see it.|',
+        '{2:  }        :r {1:^!}{3:ls}                                                                |',
+        '{2:  }     :{1:^!}{3:ls}                    - shows a directory listing                      |',
+        '{2:  } 5. {4::r }{5:^!ls}                   reads the output of the ls command and           |',
+        '{2:  } 1. List the contents of the current directory: :{1:^!}{3:ls}                          |'
+      },
+      ['!rm'] = {
+        '{2:  }        :{1:^!}{3:rm} TEST                                                             |',
+        '{2:  }     :{1:^!}{3:rm}  FILENAME          - removes file FILENAME                          |',
+      }
+    }
+
+    local screen = Screen.new(80, 1)
+    screen:set_default_attr_ids({
+      [1] = { foreground = Screen.colors.Brown, bold = true, background = Screen.colors.Yellow1 },
+      [2] = { foreground = Screen.colors.DarkBlue, background = Screen.colors.Gray },
+      [3] = { background = Screen.colors.Yellow },
+      [4] = { underline = true, foreground = tonumber('0x0088ff') },
+      [5] = { underline = true, foreground = tonumber('0x0088ff'), background = Screen.colors.Yellow },
+    })
+    screen:attach()
+
+    for search_word, search_matches in pairs(test_table) do
+      feed('gg0<CR>')
+      for _, expected_line in ipairs(search_matches) do
+        feed('/' .. search_word .. '<CR>')
+        screen:expect(expected_line)
```

Comment:
why call expect() on each line instead of one expect() call that gives the full screen? That might be slowing down the test.

Reply:
yes, but that would result in a verbose test. Just position the screen to some part of the buffer that is representative, we don't need to test the entire buffer.

Reply:
I've created a test that checks the whole 300 lines that has the transformed content and it seems to be running virtually the same speed on my machine, I pushed to see if the performance is different on CI

Reply:
That's good, but the whole buffer doesn't need to be tested. Test verbosity is also a consideration.